### PR TITLE
perf: small improve on uuid generation in doc

### PR DESCRIPTION
**Pull request by bwanglzu**
test settings: 1 million documents generation:

```python
from jina import Document

[Document() for _ in range(1000000)]
```

feature branch: 21239777 function calls (21233402 primitive calls) in 16.524 seconds

master branch: 23239839 function calls (23233463 primitive calls) in 19.815 seconds

we saved 2million function calls and roughly 17% time saving.

old & new uuid looks a bit different: 

D0364F55-4D62-490A-AB1D-9749A233AC7A

### CODE REVIEW

**Comment by davidbp** [comment]
Just note that the output will be formatted  differently. From `'707a71ee-4883-11ec-a415-787b8ab3f5de'` to '707a71ee488311eca415787b8ab3f5de'. To me it still makes sense.

File: jina/types/document/__init__.py [hunk 1]
```diff
@@ -265,7 +266,7 @@ def _update_doc(d: Dict):
             ) from ex
 
         if self._pb_body.id is None or not self._pb_body.id:
-            self.id = random_identity(use_uuid1=True)
```
Comment (JoanFM) [inline jina/types/document/__init__.py:268]: please remove this function from the `codebase` if it nos used

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Comment by bwanglzu** [comment]
we `str` conversion, it is:
```
21239914 function calls (21233528 primitive calls) in 17.544 seconds
```

**Comment by tadejsv** [comment]
Can you also test by using `uuid4`? This is actually the one used now

**Comment by hanxiao** [comment]
nice catch!

but then i would suggest refactoring `random_identity` to use `.hex`, instead of making this exception for Document constructor.

**Comment by JoanFM** [comment]
> nice catch!
> 
> but then i would suggest refactoring `random_identity` to use `.hex`, instead of making this exception for Document constructor.

I would do that, but at the same time keep the `.hex` here, saving the function call and the `if else` can be good

**Comment by bwanglzu** [comment]
okay now I break a lot of distributed test

**Comment by JoanFM** [comment]
> okay now I break a lot of distributed test

no pain, no gain

**Comment by bwanglzu** [comment]
> > okay now I break a lot of distributed test
> 
> no pain, no gain

:sob:

**Review by JoanFM — APPROVED** [review]

**Pull request merged**